pipeline-runner/R/qc-6-integrate_scdata.R

```diff
@@ -268,3 +287,12 @@ colorObject <- function(data) {
   }
   return(data)
 }
+
+estimate_npcs <- function(scdata, var_threshold, max_npcs) {
+  # estimates the number of PCs to use in data integration and embeddings,
+  # using accumulated explained variance
+  eigValues <- (scdata@reductions$pca@stdev)^2
+  varExplained <- eigValues / sum(eigValues)
+  npcs <- min(which(cumsum(varExplained) >= var_threshold))
+  min(npcs, max_npcs, na.rm = TRUE)
```

Comment:
```suggestion
  eig_values <- (scdata@reductions$pca@stdev)^2
  var_explained <- eig_values / sum(eig_values)
  npcs <- min(which(cumsum(var_explained) >= var_threshold))
  min(npcs, max_npcs, na.rm = TRUE)
```

Reply:
- The norm at least right now is snake case for variable names (functions also generally snake case in pipeline).

- What is the rationale for upper threshold on `npcs`?
- How did you decide on this method for choosing npcs? It doesn't seem to be among the methods suggested either in the [Bioconductor world](http://bioconductor.org/books/3.14/OSCA.advanced/dimensionality-reduction-redux.html#using-the-elbow-point) or [Seurat vignettes](https://satijalab.org/seurat/articles/pbmc3k_tutorial.html#determine-the-dimensionality-of-the-dataset)